torch/csrc/distributed/rpc/init.cpp

```diff
@@ -475,9 +475,22 @@ PyObject* rpc_init(PyObject* /* unused */) {
   py::class_<TensorPipeRpcBackendOptions>(
       module, "TensorPipeRpcBackendOptions", rpcBackendOptions)
       .def(
-          py::init<float, std::string>(),
+          py::init<
+              int,
+              optional<std::vector<std::string>>,
+              optional<std::vector<std::string>>,
+              float,
+              std::string>(),
+          py::arg("num_worker_threads") = kDefaultNumWorkerThreads,
+          py::arg("_transports") = optional<std::vector<std::string>>(),
+          py::arg("_channels") = optional<std::vector<std::string>>(),
           py::arg("rpc_timeout") = kDefaultRpcTimeoutSeconds,
-          py::arg("init_method") = kDefaultInitMethod);
+          py::arg("init_method") = kDefaultInitMethod)
+      .def_readwrite(
+          "num_worker_threads", &TensorPipeRpcBackendOptions::numWorkerThreads);
+
+  module.attr("_DEFAULT_NUM_WORKER_THREADS") =
+      py::cast(kDefaultNumWorkerThreads);
```

Comment:
What do you mean? For a start, they have different values: NUM_SEND_RECV threads is 4 by default, NUM_WORKER_THREADS is 16. Also, they control different aspects. I am not so familiar with the ProcessGroup agent but I think those threads deal both with I/O (receiving and sending) and with the execution of the requests. With TensorPipe, the worker threads just execute the requests, while separate threads deal with I/O. (There's no controls for those threads yet, because they are specific to transports/channels, and we don't have a way to configure options for them yet).

Reply:
I see